scheduler-worker/src/test/java/gov/nasa/jpl/aerie/scheduler/worker/services/SchedulingIntegrationTests.java

```diff
@@ -305,6 +308,234 @@ export default () => Goal.CoexistenceGoal({
+                null,
+                true
+            ),
+            new ActivityDirective(
+                Duration.MINUTE.times(5),
+                "BiteBanana",
+                Map.of("biteSize", SerializedValue.of(1)),
+                null,
+                true
+            ),
+            expectedSatisfactionAct,
+            new ActivityDirective(
+                Duration.MINUTE.times(10),
+                "GrowBanana",
+                Map.of(
+                    "quantity", SerializedValue.of(2),
+                    "growingDuration", SerializedValue.of(Duration.MINUTE.times(2).in(MICROSECOND))
+                ),
+                null,
+                true
+            )
+        ),
+        List.of(new SchedulingGoal(new GoalId(0L), """
+          export default () => Goal.CoexistenceGoal({
+            forEach: ActivityExpression.ofType(ActivityTypes.BiteBanana),
+            activityFinder: ActivityExpression.build(ActivityTypes.GrowBanana, {quantity: 1}),
+            activityTemplate: (interval) => ActivityTemplates.GrowBanana({quantity: 1, growingDuration: Temporal.Duration.from({minutes:1}) }),
+            startsAt: TimingConstraint.singleton(WindowProperty.END).plus(Temporal.Duration.from({ minutes : 5}))
+          })
+          """, true)),
+        PLANNING_HORIZON);
+
+    assertEquals(1, results.scheduleResults.goalResults().size());
+    final var goalResult = results.scheduleResults.goalResults().get(new GoalId(0L));
+
+    assertTrue(goalResult.satisfied());
+    assertEquals(1, goalResult.createdActivities().size());
+    for (final var activity : goalResult.createdActivities()) {
+      assertNotNull(activity);
+    }
+    assertEquals(2, goalResult.satisfyingActivities().size());
+    var foundSatisfactionAct = false;
+    for (final var activity : goalResult.satisfyingActivities()) {
+      assertNotNull(activity);
+      final var element = results.idToAct.get(new ActivityDirectiveId(-activity.id()));
+      if(element != null && element.equals(expectedSatisfactionAct)){
+        foundSatisfactionAct = true;
+      }
+    }
+    assertTrue(foundSatisfactionAct);
+    final var planByActivityType = partitionByActivityType(results.updatedPlan());
+    final var growBananas = planByActivityType.get("GrowBanana");
+    assertEquals(3, growBananas.size());
+    final var planByTime = partitionByStartTime(results.updatedPlan());
+    assertEquals(2, planByTime.get(MINUTE.times(10)).size());
+    var lookingFor = false;
+    final var expectedCreation = new SerializedActivity("GrowBanana",
+                                            Map.of("quantity", SerializedValue.of(1),
+                                                   "growingDuration", SerializedValue.of(MINUTES.in(MICROSECOND))));
+    for(final var actAtTime10: planByTime.get(MINUTE.times(10))){
+      if(actAtTime10.serializedActivity().equals(expectedCreation)){
+        lookingFor = true;
+      }
+    }
+    assertTrue(lookingFor);
```

Comment:
Not this PR, but it's kinda irksome that the equality check has to be done like this because `MockMerlinService` drops ActivityIds

Reply:
True